MAINT - Defer FontAwesome script to improve Lighthouse performance score (#2209)
pydata/pydata-sphinx-theme: docs/user_guide/header-links.rst

In order to make use of the full feature set of ``.svg`` images provided by HTML you will need
to set up the ``.svg`` to be used as a FontAwesome type icon. This is a fairly straightforward process:

> #. Copy `custom-icon.js <https://github.com/pydata/pydata-sphinx-theme/blob/main/docs/_static/custom-icon.js>`_

Author: I wanted to make this an internal link but the following syntax did not work:

```
:doc:`custom-icon.js </_static/custom-icon.js>`
```

   into an appropriate directory of your documentation source (typically ``source/js``). In that file you will
   find a list of custom icon definitions. Each definition looks like this:

   .. code:: javascript

     {
       prefix: "fa-custom",
       iconName: "pypi",
       icon: [
         17.313, // viewBox width
         19.807, // viewBox height
         [], // ligature
         "e001", // unicode codepoint - private use area
         "m10.383 0.2-3.239 ...", // svg path
       ],
     }

#. You will clear the list of icon definitions and add your own using the same pattern as above:

   #.  Keep the value of ``prefix`` as "fa-custom"
   #.  Set the value of ``iconName``  to a short, sensible, lowercase name for your icon
   #.  Change the view box height and width to match that of your icon
   #.  For each custom icon you define in the list, increment the Unicode codepoint by 1: ``"e001"``, ``"e002"``, ``"e003"``...
   #.  Replace the SVG path string with the path that defines your custom icon

#. In ``conf.py``, add the relative path from your source directory to the custom javascript file:

   .. code:: python

      html_js_files = [
         ...

>          ("js/pypi-icon.js", {"defer": "defer"}),

Author: Going to update this to be "js/custom-icons.js", because I think that will make the instructions a bit easier to follow

         ...
      ]

>    .. versionchanged:: v0.17.0

Reviewer: No need to add as `block-release`, the PR seems pretty much ready now.

In order to make use of the full feature set of ``.svg`` images provided by HTML you will need
to set up the ``.svg`` to be used as a FontAwesome type icon. This is a fairly straightforward process:

#. Copy `custom-icons.js`_ into an appropriate directory of your documentation
   source (typically ``source/js``). In that file you will find a list of custom
   icon definitions. Each definition looks like this:

   .. code:: javascript

     {
       prefix: "fa-custom",
       iconName: "pypi",
       icon: [
         17.313, // viewBox width
         19.807, // viewBox height
         [], // ligature
         "e001", // unicode codepoint - private use area
         "m10.383 0.2-3.239 ...", // svg path
       ],
     }

> #. You will clear the list of icon definitions and add your own using the same pattern as above:

Reviewer: ```suggestion
#. Remove the existing list of icon definitions and add your own using the same pattern as above:
```
